lib/src/app.dart

```diff
@@ -1,54 +1,21 @@
 import 'package:beamer/beamer.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
-import 'package:provider/provider.dart';
-import 'package:the_monkslab_web/src/features/_index.dart';
+import 'package:the_monkslab_web/src/core/_index.dart';
 
 class AppPage extends StatelessWidget {
-  AppPage({super.key});
-
-  final routerDelegate = BeamerDelegate(
-    locationBuilder: RoutesLocationBuilder(
-      routes: {
-        // Return either Widgets or BeamPages if more customization is needed
-        '/': (context, state, data) => const HomePage(),
-        '/archive': (_, __, ___) => const ArchivePage(),
-        '/course/:path': (_, state, __) {
-          final path = state.pathParameters['path']!;
-          return CoursePage(path);
-        },
-        '/article/:path': ((_, state, __) {
-          final path = (state.pathParameters['path']!);
-          return ArticlePage(path: path);
-        }),
-      },
-    ),
-  );
+  const AppPage({super.key});
 
   @override
   Widget build(BuildContext context) {
-    return Consumer<LocaleProvider>(
-      builder: (context, provider, snapshot) {
-        return MaterialApp.router(
-          title: 'The MonksLab',
-          routeInformationParser: BeamerParser(),
-          routerDelegate: routerDelegate,
-          localizationsDelegates: AppLocalizations.localizationsDelegates,
-          supportedLocales: AppLocalizations.supportedLocales,
-          locale: provider.locale,
-        );
-      },
+    return MaterialApp.router(
+      title: 'The MonksLab',
+      routeInformationParser: BeamerParser(),
+      routerDelegate: routerDelegate,
+      localizationsDelegates: AppLocalizations.localizationsDelegates,
+      supportedLocales: AppLocalizations.supportedLocales,
+      locale: context.watch<LocaleCubit>().state.locale,
```

Comment:
Bloc watch to check locale changes

Reply:
Oh! This way whenever we switch the cubit, the whole thing changes?

Reply:
We, It rebuilds the whole Material app and under.

Reply:
Cooo!